src/components/Integrations/Shopify/Shopify.js

```diff
@@ -99,6 +81,31 @@ const Shopify = ({ intl, dependencies: { shopifyClient } }) => {
     </>
   );
 
+  const getData = useCallback(async () => {
+    const result = await shopifyClient.getShopifyData();
+    if (!result.success) {
+      setError(<FormattedHTMLMessage id="validation_messages.error_unexpected_HTML" />);
+    } else if (result.value.length) {
+      setConnectionData({ shops: result.value, isConnected: true });
+      setIsConnecting(false);
+      setError(null);
+    } else {
+      setConnectionData({ shops: result.value, isConnected: false });
+      setError(null);
+    }
+    if (!isConnecting) {
+      setIsLoading(false);
+    }
+  }, [isConnecting, shopifyClient]);
+
+  useEffect(() => {
```

Comment:
Why is _useEffect_ necessary?

Reply:
Thanks to _useEffect_, the content is executed after the first rendering, probably it is fine, but, why?

Reply:
As you've said I need this to execute the first time, else I have to wait for 20 seconds with a blank screen. O_o

Reply:
We fixed this changing the behaviour for `useInterval` hook adding the possibility to execute first time.